tests/integ/test_pandas_to_df.py

```diff
@@ -295,6 +304,90 @@ def test_create_dataframe_from_pandas(session):
     # assert_frame_equal(results, pd, check_dtype=False)
 
 
+def test_create_dataframe_from_pandas_with_schema(session):
```

Comment:
sorry I should mention it in my previous comments, but let's try binary and decimal types? We have such tests on these two types for create_dataframe from native python values.

Reply:
I added test for decimal types. 

Today `write_pandas` does not work great with `binary` type as I will hit the following if I created pandas DF with `b"t1"`:

`Expression type does not match column data type, expecting BINARY(8388608) but got VARIANT for column SHOE_MODEL_BINARY`

Looks like a bug and we should use `TO_BINARY` for our generated `copy_into_sql` [here](https://github.com/snowflakedb/snowflake-connector-python/blob/0afbf902b020c99d2db47e24aa181ad2085c421c/src/snowflake/connector/pandas_tools.py#L268-L273)? cc @sfc-gh-mkeller  If that is the case then I could create a ticket so we could work on that separately, and in the meanwhile create another TODO to add test for binary type.

Reply:
Apparently there was a long discussion about binary not working since python connector 2.3.7. Currently this is blocked by snowflake internal feature tracked in https://snowflakecomputing.atlassian.net/browse/SNOW-204246